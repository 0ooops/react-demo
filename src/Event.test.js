import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'
import EventPage from './Event';

configure({ adapter: new Adapter() });

it('event page renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<EventPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('event page snapshot test', () => {
  const component = renderer.create(<EventPage />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('event page title renders properly', () => {
  const title = 'Recent Events'
  const pageWrapper = shallow(<EventPage />);
  expect(pageWrapper.find('h1').text()).toEqual(title);
});

/* more to added for testing each component */