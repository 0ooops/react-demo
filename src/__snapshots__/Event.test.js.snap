// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`event page snapshot test 1`] = `
<div
  className="container"
>
  <h1
    className="event-page-title"
  >
    Recent Events
  </h1>
  <div>
    Loading...
  </div>
</div>
`;
